test: E2E

yassine04e opened the pull request.

Unsaved2 commented on app/src/main/java/com/github/se/travelpouch/ui/authentication/SignInWithPassword.kt:
```diff
@@ -0,0 +1,87 @@
+                      )
+                      .show()
+                })
+          }) {
```
It would be better to enable the button only if there is a password and an email

yassine04e replied:
Yes, you are right

Unsaved2 reviewed (changes requested):
Great Work ! We have now a End to end test. I would have preferred that we use the Firebase emulator to do the end to end but I think you had problems with it so using Hilt is a very good way to do it. 

Some suggestion in addition : 
1. Add Appmodule in a folder `di` (Dependency Injection)
2. Add `ProfileRepositoryMock`, `TravelRepositoryMock` in a folder mock
3. I have the impression when we click on the Google sign in the button behaves strangely 
4. The sign button should be less visible because it is not the main option for sign in
5. The screen for the Sign in with password can be maybe prettier ?

yassine04e commented on app/build.gradle.kts:
```diff
@@ -172,14 +174,16 @@ dependencies {
-    // Kotlin
-    implementation(libs.kotlinx.coroutines.guava)
+    testImplementation("com.google.dagger:hilt-android-testing:2.51.1")
+    kaptTest("com.google.dagger:hilt-android-compiler:2.51.1")
```
Can you be more spicific on what I should do with the second part ?

Unsaved2 replied:
I check and apparently there is no need to put the second part

yassine04e commented on app/src/androidTest/java/com/github/se/travelpouch/endtoend/EndToEndTest.kt:
```diff
@@ -0,0 +1,123 @@
+
+  @Test
+  fun verifyUserFlowForTravelCreation() =
+      runTest(timeout = Duration.INFINITE) {
```
Yes, you are right

yassine04e commented on app/src/androidTest/java/com/github/se/travelpouch/endtoend/EndToEndTest.kt:
```diff
@@ -0,0 +1,123 @@
+
+        // assert that login screen is displayed
+        composeTestRule.onNodeWithTag("appLogo").assertIsDisplayed()
+        composeTestRule.onNodeWithTag("welcomText").assertIsDisplayed()
```
Well, Antoine made the typo. I don't want to modify it because I am not sure if he will use the former tag.

Unsaved2 replied:
I think we can correct it, but it's up to you

yassine04e commented on app/src/androidTest/java/com/github/se/travelpouch/ui/authentication/TestSignInWithEmailAndPassword.kt:
```diff
@@ -0,0 +1,80 @@
+package com.github.se.travelpouch.ui.authentication
+
+import androidx.compose.ui.test.junit4.createComposeRule
```
Yes. I will put it in a the unit test folder

yassine04e commented on app/src/main/AndroidManifest.xml:
```diff
@@ -3,6 +3,7 @@
     xmlns:tools="http://schemas.android.com/tools">
 
     <application
+        android:name=".TravelPouchApp"
```
Honestly I am not sure. I just followed this video : https://www.youtube.com/watch?v=bbMsuI2p1DQ&t=1419s

Unsaved2 replied:
Ok I think it is correct

yassine04e commented on app/src/androidTest/java/com/github/se/travelpouch/ui/authentication/TestSignInWithEmailAndPassword.kt:
```diff
@@ -0,0 +1,80 @@
+
+  @Test
+  fun testSignInWithEmailAndPasswordWorksCorrectly() =
+      runTest(timeout = Duration.INFINITE) {
```
Yes, I did it

yassine04e commented on app/src/androidTest/java/com/github/se/travelpouch/ui/authentication/TestSignInWithEmailAndPassword.kt:
```diff
@@ -0,0 +1,80 @@
+  @Test
+  fun testSignInWithEmailAndPasswordIfTaskFails() =
+      runTest(timeout = Duration.INFINITE) {
+        val firebaseAuthSerice = FirebaseAuthenticationService(mockFirebaseAuth)
```
Yes, I will correct it

yassine04e commented on app/src/androidTest/java/com/github/se/travelpouch/ui/authentication/TestSignInWithEmailAndPassword.kt:
```diff
@@ -0,0 +1,80 @@
+        verify(task).addOnCompleteListener(onCompleteListenerCaptor.capture())
+        onCompleteListenerCaptor.firstValue.onComplete(task)
+
+        assert(!onSuccessCalled)
```
Ok, you are right

yassine04e commented on app/src/main/java/com/github/se/travelpouch/Database.kt:
```diff
@@ -0,0 +1,7 @@
+package com.github.se.travelpouch
+
```
I added this to simulate the firebase database

Unsaved2 replied:
ok I understand better

yassine04e commented on app/src/main/java/com/github/se/travelpouch/AuthenticationService.kt:
```diff
@@ -0,0 +1,50 @@
+package com.github.se.travelpouch
+
```
We added an adapter to be able to mock the authentication service. So that we can mock it throught an interface it shares with a mock authentication service

Unsaved2 replied:
ok make sense

yassine04e commented on app/src/main/java/com/github/se/travelpouch/MainActivity.kt:
```diff
@@ -55,79 +64,91 @@ class MainActivity : ComponentActivity() {
-      composable(Screen.AUTH) {
-        SignInScreen(navigationActions, profileModelView, listTravelViewModel)
-      }
+    Log.d("ENDTOEND-FINAL", "after all viewmodels")
```
I will remove thsi log

yassine04e commented on app/src/main/java/com/github/se/travelpouch/model/travels/TravelRepositoryMock.kt:
```diff
@@ -0,0 +1,68 @@
+      onSuccess: (Profile?) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    TODO("Not yet implemented")
```
Technically these functions are not used so no need to implement them. Later, we will have to implenment them. But for new they are not usefull (and I cannot remove  them since they come from an interface the class implements)

yassine04e commented on app/src/main/java/com/github/se/travelpouch/ui/authentication/SignInWithPassword.kt:
```diff
@@ -0,0 +1,87 @@
+                  Log.d("SignInScreen", "User signed in: ${user?.displayName}")
+
+                  val job =
+                      GlobalScope.launch {
```
Ok

yassine04e commented on app/src/main/java/com/github/se/travelpouch/model/profile/ProfileModelView.kt:
```diff
@@ -18,19 +17,21 @@ import kotlinx.coroutines.flow.asStateFlow
-    val Factory: ViewModelProvider.Factory =
-        object : ViewModelProvider.Factory {
-          @Suppress("UNCHECKED_CAST")
-          override fun <T : ViewModel> create(modelClass: Class<T>): T {
```
Now the factory is useless since we are using hilt injection. So at the creation of the view model, we use injection, which can take two different values depending of if we use tests or app

yassine04e commented:
To answer to your main comment : 
1) Yes, I added the di folder.
2) I think we don't need a mock module since the mock repositories are considered as the model part of our application and they just implement an interface of a specific repository.
3) No need, we can have two authentication options. 
4) Yes, I corrected it.

Unsaved2 commented on app/src/main/java/com/github/se/travelpouch/model/travels/ListTravelViewModel.kt:
```diff
@@ -37,16 +38,16 @@ open class ListTravelViewModel(private val repository: TravelRepository) : ViewM
-        }
-  }
+  //  // create factory
+  //  companion object {
```
yes you are right
